include/bluetooth/bluetooth.h

```diff
@@ -404,6 +537,141 @@ int bt_le_adv_update_data(const struct bt_data *ad, size_t ad_len,
+	/** Advertiser timeout (N * 10 ms), Application will be notified by the
+	 *  advertiser sent callback. Set to zero for no timeout.
+	 *
+	 *  If privacy :option:`CONFIG_BT_PRIVACY` is enabled then the timeout
+	 *  must be less than :option:`CONFIG_BT_RPA_TIMEOUT`.
+	 */
+	u16_t timeout;
+	/** Number of advertising events. Application will be notified by the
+	 *  advertiser sent callback. If used together with timeout then
+	 *  the condition that happens first will stop the advertiser.
+	 *  Set to zero for no limit.
+	 */
+	u8_t  num_events;
+};
+
+/** Start advertising with the given advertising set
+ *
+ *  Start advertising with a specific advertising set.
+ *  If both timeout and advertising event limit is given then the advertising
+ *  set will be stopped once one of them is reached.
+ *
+ *  @param adv    Advertising set object.
+ *  @param param  Advertise start parameters.
+ */
+int bt_le_ext_adv_start(struct bt_le_ext_adv *adv,
+			struct bt_le_ext_adv_start_param *param);
+
+/** Stop advertising with the given advertising set
+ *
+ *  Stop advertising with a specific advertising set. When using this function
+ *  the advertising sent callback will not be called.
+ *
+ *  @param adv Advertising set object.
+ *
+ *  @return Zero on success or (negative) error code otherwise.
+ */
+int bt_le_ext_adv_stop(struct bt_le_ext_adv *adv);
+
+/** Set advertising set advertising or scan response data.
+ *
+ *  Set advertisement data or scan response data. If the advertising set is
+ *  currently advertising then the advertising data will be updated in
+ *  subsequent advertising events.
+ *
+ *  If the advertising set has been configured to send advertising data on the
+ *  primary advertising channels then the maximum data length is
+ *  @ref BT_GAP_ADV_MAX_ADV_DATA_LEN bytes.
+ *  If the advertising set has been configured for extended advertising,
+ *  then the maximum data length is defined by the controller with the maximum
+ *  possible of @ref BT_GAP_ADV_MAX_EXT_ADV_DATA_LEN bytes.
+ *
+ *  @note Not all scanners support extended data length advertising data.
+ *
+ *  @param adv     Advertising set object.
+ *  @param ad      Data to be used in advertisement packets.
+ *  @param ad_len  Number of elements in ad
+ *  @param sd      Data to be used in scan response packets.
+ *  @param sd_len  Number of elements in sd
+ *
+ *  @return Zero on success or (negative) error code otherwise.
+ */
+int bt_le_ext_adv_set_data(struct bt_le_ext_adv *adv,
+			   const struct bt_data *ad, size_t ad_len,
+			   const struct bt_data *sd, size_t sd_len);
+
+/** @brief Update advertising parameters.
+ *
+ *  Update the advertising parameters. The advertiser will have to be stopped
+ *  before the parameters can be updated.
+ *
+ *  @param adv   Advertising set object.
+ *  @param param Advertising parameters.
+ *
+ *  @return Zero on success or (negative) error code otherwise.
+ */
+int bt_le_ext_adv_update_param(struct bt_le_ext_adv *adv,
```

Comment:
`bt_le_ext_adv_update_param()` is not implemented. I assume that the user should use `bt_le_ext_adv_start()` instead? The advertiser will anyways have to be stopped

Reply:
No, `bt_le_ext_adv_update_param()` well be implemented, just haven't gotten around to it. `bt_le_ext_adv_start` does not have the advertising parameters as input.